hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/HandleWorkflow.java

```diff
@@ -609,6 +615,59 @@ private void handleUserTransaction(
         handleWorkflowMetrics.update(transactionInfo.functionality(), handleDuration);
     }
 
+    /**
+     * Returns a set of "extra" account ids that should be considered as eligible for
+     * staking rewards with the given transaction info and record builder. Needed for
+     * mono-service fidelity.
+     *
+     * <p>There are two cases.
+     * <ol>
+     *     <li>For a crypto transfer, any account explicitly given a zero amount
+     *     should be considered as eligible for staking rewards.</li>
```

Comment:
Is this because we actually don't change that account in state, so its not in modifications ? Also isn't mono-service behavior wrong ?

Reply:
Yes mono-service has "false positives" in its reward situation detection, because it is triggered by presence of the `BALANCE` property in an entity's `TransactionalLedger` change set. (As you well know. 😁)

Updated the javadoc here to reflect this.